linode_api4/objects/serializable.py

```diff
@@ -0,0 +1,85 @@
+import copy
+import inspect
+from dataclasses import dataclass
+from typing import Any, Dict, get_type_hints
+
+
+@dataclass
+class JSONObject:
+    """
+    A simple helper class for serializable API objects.
+    This is typically used for nested object values.
+
+    This class act similarly to MappedObject but with explicit
+    fields and static typing.
+    """
+
+    def __init__(self):
+        raise NotImplementedError(
+            "JSONObject is not intended to be constructed directly"
+        )
+
+    # TODO: Implement __repr__
+
+    @classmethod
+    def from_json(cls, json: Dict[str, Any]) -> "JSONObject":
```

Comment:
There is also a popular library for parsing JSON string input to an instance of dataclass, but I am not sure if we want to add one more dependency for this repo.

https://github.com/lidatong/dataclasses-json